core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -817,19 +882,70 @@ void core_mmu_get_user_va_range(vaddr_t *base, size_t *size)
 {
 	assert(user_va_idx != -1);
 
+	/*
+	 * Base address is determined by base page table level granularity
+	 * (can be 0 or 1), while address space size is always of level 1.
+	 */
 	if (base)
-		*base = (vaddr_t)user_va_idx << L1_XLAT_ADDRESS_SHIFT;
+		*base = (vaddr_t)user_va_idx << BASE_XLAT_ADDRESS_SHIFT;
 	if (size)
 		*size = BIT64(L1_XLAT_ADDRESS_SHIFT);
 }
 
+static uint64_t *core_mmu_get_user_mapping_entry(struct mmu_partition *prtn,
+						 unsigned int base_idx)
+{
+	uint64_t *tbl;
+	unsigned int n;
+#if (CORE_MMU_BASE_TABLE_LEVEL == 0)
+	uintptr_t ntbl;
+#endif
+
+	assert(prtn);
+	assert(base_idx < NUM_BASE_TABLES);
+	assert(user_va_idx != -1);
+
+	tbl = prtn->base_tables[base_idx][get_core_pos()];
+	n = user_va_idx;
+
+	/*
+	 * If base level is 0, then user mapping entry is entry 0 in the level 1
+	 * table pointed by user_va_idx.
+	 */
+#if (CORE_MMU_BASE_TABLE_LEVEL == 0)
```

Comment:
We try to avoid `#if` and `#ifdef` inside the functions. There's a bit of the mix in the source, but this is the direction where working towards.
A plain `if (CORE_MMU_BASE_TABLE_LEVEL == 0)` should work here.